subsys/power/Kconfig

```diff
@@ -17,9 +17,26 @@ config PM_CONTROL_OS_DEBUG
 	help
 	  Enable OS Power Management debugging hooks.
 
+config DEVICE_IDLE_PM
+	bool "Enable device Idle Power Management"
+	depends on DEVICE_POWER_MANAGEMENT
+	help
+	  Enable device Idle Power Management to save power.
+	  With device Idle PM enabled, devices can be suspended or
+	  resumed based on the device usage even while the CPU or
+	  system is running.
+
+config DEVPM_WORKQUEUE_STACK_SIZE
+	int "Device PM workqueue stack size"
+	depends on DEVICE_IDLE_PM
+	default 1024
```

Comment:
1024B of RAM plus maybe a few hundred more per driver... just to count users... and I was looking how to free some memory in Zephyr just a week ago.
Thankfully this can be disabled and I hope this will not become mandatory.

Reply:
This is purely a config based option. The option can be enabled only if the application wants OS based PM and device idle management.

Reply:
True, but this would also require changes in the drivers. If drivers will implement it then application would have to use it as it will become a dependency.